### ISSUE TITLE
Roslyn Source Generators not compatible with F#

### ISSUE THREAD

**Issue by pjmagee** [wingyplus/dagger-dotnet-sdk#12]
dotnet SDK != `C#`

- `dagger init -sdk csharp` - make csharp template, and pull nuget package `Dagger.SDK.dll`
- `dagger init -sdk powershell`

If we keep all the code as a self contained package, I believe user code can iun theory, be called by even Powershell?

**Comment by pjmagee** [comment]
This relates to the other issue with renaming the SDK. 
An F# solution might be possible, but it would need to use a csharp project and reference it.

**Comment by wingyplus** [comment]
It might possible to do that. I have a draft in https://github.com/wingyplus/dagger-dotnet-sdk/issues/13#issuecomment-2150378369 but need to make C# version works to make it clear on how to do next.

**Comment by pjmagee** [comment]
> It might possible to do that. I have a draft in [#13 (comment)](https://github.com/wingyplus/dagger-dotnet-sdk/issues/13#issuecomment-2150378369) but need to make C# version works to make it clear on how to do next.

Yeah, after some additional understanding, F# project can reference C# project (which is used to BUILD + Generate C# code from Roslyn generator) and after this, an F# developer could then use it... 

But i am wondering what is experience like for F# entry point with concept of Dagger functions etc... Interesting subject
